### DISCUSSION TITLE
Best way to implement Wrapping in TextArea?

### DISCUSSION THREAD

**Discussion by jspv** [Textualize/textual#3598] [Q&A]
In my app, I want long lines to soft wrap and not generate a X axis scrollbar, any thoughts on the best way to do this?  

I've started down the rabbit hole of monitoring the virtual width and was going to be inserting and tracking newlines (tracking so that I can remove them when I process the text), but this seems overly low level given that most widgets automatically wrap text to the visible area.  Is there a smarter/better way to do this?

**Comment by willmcgugan** [comment]
The TextArea currently doesn't support word wrap,  but it is in the pipeline.

Do you need the editing capability? Because if it is just a matter of displaying text there are a few options, which will word wrap automatically.

**Reply by jspv** [reply]
Unfortunately yes, I'm using it as a multi-line input widget which could contain content with hard carriage returns - intent is to soft wrap long lines to the available space and not show a scrollbar.
